src/figuresClasses.ts

```diff
@@ -1,11 +1,88 @@
-export interface Figure {}
+function roundDown(number: number, decimals: number): number {
+  const decims = decimals || 0;
 
-export class Triangle implements Figure {}
+  return Math.floor(number * Math.pow(10, decims)) / Math.pow(10, decims);
+}
+
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c];
 
-export class Circle implements Figure {}
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Side of triangle can not be <= 0');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. A more descriptive error message would be helpful, such as 'Each side of the triangle must be greater than 0.'